plugins/samples/enable_recaptcha/plugin.rs

```diff
@@ -0,0 +1,221 @@
+// Copyright 2025 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// [START serviceextensions_plugin_enable_recaptcha]
+
+// Warning: This plugin simply shows that by adding scripts, one could enable a
+// reCAPTCHA challenge. This is not a replacement for reading reCAPTCHA
+// documentaion or following all user guide instructions. Please follow official
+// reCAPTCHA documentation at https://developers.google.com/recaptcha.
+use lol_html::html_content::ContentType;
+use lol_html::*;
+use proxy_wasm::traits::*;
+use proxy_wasm::types::*;
+use std::cell::RefCell;
+use std::error::Error;
+use std::rc::Rc;
+
+proxy_wasm::main! {{
+    proxy_wasm::set_log_level(LogLevel::Trace);
+    proxy_wasm::set_root_context(|_| -> Box<dyn RootContext> {
+      Box::new(MyRootContext {
+          recaptcha_key: Rc::new(RefCell::new(String::new())),
+      })
+    });
+}}
+
+struct MyRootContext {
+    recaptcha_key: Rc<RefCell<String>>,
+}
+
+impl Context for MyRootContext {}
+
+impl RootContext for MyRootContext {
+    fn on_configure(&mut self, _: usize) -> bool {
+        if let Some(config) = self.get_plugin_configuration() {
+            // Config file contains only the recaptcha key.
+            // Use of .unwrap() is fine here, since failure in on_configure
+            // should result in plugin crash.
+            let config_lines = String::from_utf8(config).unwrap();
```

Comment:
I don't know if this is already doing this or not but it would be good to trim whitespace.

Reply:
I intentionally kept the handling of config data as bare-bones as possible. Users can trim whitespace if they find it necessary, but it isn't needed in this instance.